openedx/core/djangoapps/site_configuration/middleware.py

```diff
@@ -38,17 +38,32 @@ def _set_cookie_wrapper(key, value='', max_age=None, expires=None, path='/', dom
                 if key == configuration_helpers.get_value('SESSION_COOKIE_NAME', settings.SESSION_COOKIE_NAME):
                     domain = session_cookie_domain
 
-                # then call down into the normal Django set_cookie method
-                return response.set_cookie_wrapped_func(
-                    key,
-                    value,
-                    max_age=max_age,
-                    expires=expires,
-                    path=path,
-                    domain=domain,
-                    secure=secure,
-                    httponly=httponly
-                )
+                # samesite flag was added in django 2.1, so only pass it in for django 2.1 or higher
+                if django.VERSION >= (2, 1):
+                    # then call down into the normal Django set_cookie method
+                    return response.set_cookie_wrapped_func(
+                        key,
+                        value,
+                        max_age=max_age,
+                        expires=expires,
+                        path=path,
+                        domain=domain,
+                        secure=secure,
+                        httponly=httponly,
+                        samesite=samesite
+                    )
+                else:
+                    # then call down into the normal Django set_cookie method
+                    return response.set_cookie_wrapped_func(
+                        key,
+                        value,
+                        max_age=max_age,
+                        expires=expires,
+                        path=path,
+                        domain=domain,
+                        secure=secure,
+                        httponly=httponly,
+                    )
```

Comment:
To make this more readable, can you create the kwargs as a dict and then add the additional kwarg only for Django2.1+.  This way, there would be only a single line difference between Django versions instead of 20-line difference.

So the final call would be:
```
return response.set_cookie_wrapped_func(key, value, **kwargs)
```

Reply:
that is good idea, will change to match this